docs/source/use-cases.rst

```diff
@@ -0,0 +1,375 @@
+Use Cases
+=========
+
+Dask is a versatile tool that supports a variety of workloads.  This page
+contains brief and illustrative examples for how people use Dask in practice.
+This page emphasizes breadth and hopefully inspires readers to find new ways
+that Dask can serve them beyond their original intent.
+
+Overview
+--------
+
+Dask algorithms cases can be roughly divided in the following two categories:
+
+1.  Large NumPy/Pandas/Lists with :doc:`dask.array<array>`,
+    :doc:`dask.dataframe<dataframe>`, :doc:`dask.bag<bag>` to analyze large
+    datasets with familiar techniques.  This is similar to Databases, Spark_,
+    or big array libraries.
+2.  Ad-hoc task scheduling.  You submit a graph of functions that depend on
+    each other for custom workloads.  This is similar to Luigi_, Airflow_,
+    Celery_, or Makefiles_.
+
+Most people today approach Dask assuming it is a framework like Spark, designed
+for the first use case around large collections of uniformly shaped  data.
+However, many of the more productive and novel use cases fall into the second
+category, using Dask to parallelize custom workflows.
+
+Dask compute environments can be divided into the following two categories:
+
+1.  Single machine parallelism with threads or processes:  The Dask
+    single-machine scheduler leverages the full CPU power of a laptop or a
+    large workstation and changes the space limitation from "fits in memory" to
+    "fits on disk".  This scheduler is simple to use and doesn't have the
+    computational or conceptual overhead of most "big data" systems.
+2.  Distributed cluster parallelism on multiple nodes:  The Dask distributed
+    scheduler coordinates the actions of multiple machines on a cluster.  It
+    scales anywhere from a single machine to a thousand machines, but not
+    significantly beyond.
+
+The single machine scheduler is useful to more individuals (more people have
+personal laptops than have access to clusters) and probably accounts for 80+% of
+the use of Dask today.  The distributed machine scheduler is useful to larger
+organizations like universities, research labs, or private companies.
+
+.. _Airflow: http://airflow.incubator.apache.org/
+.. _Luigi: https://luigi.readthedocs.io/en/latest/
+.. _Celery: http://www.celeryproject.org/
+.. _Spark: https://spark.apache.org/
+.. _Makefiles: https://en.wikipedia.org/wiki/Make_(software)
+
+Below we give specific examples of how people use Dask.  We start with large
+NumPy/Pandas/List examples because they're somewhat more familiar to people
+looking at "big data" frameworks.  We then follow with ad-hoc scheduling
+examples, which tend to be applicable more often, and are arguably a bit more
+interesting.
+
+Collection Examples
+-------------------
+
+Dask contains large parallel collections for n-dimensional arrays (similar to
+NumPy), dataframes (similar to Pandas), and lists (similar to PyToolz or
+PySpark.)
```

Comment:
I learned a bit of grammar today.  I thought the rule was to always put the period inside.